packages/protocol/src/testing/mock.ts

```diff
@@ -0,0 +1,39 @@
+import { Schema } from "@effect/schema";
+import { StreamConfig } from "../config";
+import * as proto from "../proto";
+
+export const MockFilter = Schema.transform(
+  Schema.Uint8ArrayFromSelf,
+  Schema.Struct({}),
+  {
+    strict: false,
+    decode(value) {
+      return proto.testing.MockFilter.decode(value);
+    },
+    encode(value) {
+      return proto.testing.MockFilter.encode(value).finish();
+    },
+  },
+);
+
+export type MockFilter = typeof MockFilter.Type;
+
+export const MockBlock = Schema.transform(
+  Schema.Uint8ArrayFromSelf,
+  Schema.Struct({
+    block_number: Schema.BigIntFromSelf,
```

Comment:
Notice that the protobuf library automatically converts from `snake_case` to `camelCase` so this field is actually `blockNumber`.

I believe after that change you won't need `strict: false` anymore.